sdk/identity/azure-identity/azure/identity/_credentials/azure_powershell.py

```diff
@@ -41,15 +41,25 @@
     $params['TenantId'] = $tenantId
 }}
 
-$useSecureString = $m.Version -ge [version]'2.17.0'
-if ($useSecureString) {{
+if ($m.Version -ge [version]'2.17.0' -and $m.Version -lt [version]'5.0.0') {{
     $params['AsSecureString'] = $true
 }}
 
 $token = Get-AzAccessToken @params
 $tokenValue = $token.Token
-if ($useSecureString) {{
-    $tokenValue = $tokenValue | ConvertFrom-SecureString -AsPlainText
+if ($tokenValue -is [System.Security.SecureString]) {{
+    if ($PSVersionTable.PSVersion.Major -lt 7) {{
+        try {{
+            $ssPtr = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($tokenValue)
```

Comment:
Based on [the guidance](https://learn.microsoft.com/en-us/powershell/azure/migrate-az-14.0.0?view=azps-14.1.0#azaccounts), we can have the variable outside the try block
```suggestion 
        $ssPtr = [System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($tokenValue)
        try {{
```